### PR TITLE
Doc review 2023 03 round 01

### PR DESCRIPTION
**Pull request by gonuke**
## Description
First round of incremental documentation improvements in Spring 2023.

## Motivation and Context
Documentation has become confusingly out of date and needs to be improved.

This round focuses on the home page and installation info.

### CODE REVIEW

File: doc/install/dependencies.rst [hunk 1]
```diff
@@ -79,45 +79,33 @@ Redhat linux users can do likewise with:
...
-                       --disable-debug \
-                       --with-hdf5=$HOME/dagmc_bld/HDF5 \
-                       --prefix=$HOME/dagmc_bld/MOAB
+    $ cmake ../moab -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
+              -DENABLE_PYMOAB=ON \
+              -DENABLE_BLASLAPACK=OFF \
+              -DENABLE_FORTRAN=OFF \
+              -DCMAKE_INSTALL_PREFIX=$HOME/dagmc_bld/MOAB \
+              -DCMAKE_C_COMPILER=${CC} \
+              -DCMAKE_CXX_COMPILER=${CXX} \
```
Comment (pshriwise) [inline doc/install/dependencies.rst]:
```suggestion
```

File: doc/install/index.rst [hunk 2]
```diff
@@ -1,17 +1,22 @@
 DAGMC installation overview
 ===========================
 
-This document describes the DAGMC installation process. Before you begin, you
-should know that the following will be required:
+This document describes the DAGMC installation process. 
+
+If you plan to use OpenMC_, you can follow their installation instructions with
+DAGMC built-in.
```
Comment (pshriwise) [inline doc/install/index.rst]: Might be good to link to the instructions here: https://docs.openmc.org/en/stable/quickinstall.html

File: doc/install/dependencies.rst [hunk 3]
```diff
@@ -79,45 +79,33 @@ Redhat linux users can do likewise with:
 MOAB installation
 ~~~~~~~~~~~~~~~~~
 
-As of DAGMC version 3.1, MOAB version 5.1.0 or higher is required. The following
+As of DAGMC version 3.2, MOAB version 5.3.0 or higher is required. The following
```
Comment (pshriwise) [inline doc/install/dependencies.rst]: Why is this again? My recollection is that we could still build the latest DAGMC with MOAB v5.1.0.

**Review by pshriwise — CHANGES_REQUESTED** [review]
Just a couple of small comments/questions from me. Thanks for kicking this effort off, @gonuke!

File: doc/install/dependencies.rst (hunk 3 above)
Comment (gonuke) [reply doc/install/dependencies.rst]: Good question... it may be <5.3.0 ? I'll check...

Comment (pshriwise) [reply doc/install/dependencies.rst]: Just built DAGMC develop and v3.2 against MOAB v5.1.0 and didn't run into any issues. Maybe we can leave this for now.

File: doc/install/dependencies.rst [hunk 4]
```diff
@@ -79,45 +79,31 @@ Redhat linux users can do likewise with:
...
-                       --disable-debug \
-                       --with-hdf5=$HOME/dagmc_bld/HDF5 \
-                       --prefix=$HOME/dagmc_bld/MOAB
+    $ cmake ../moab -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
+              -DENABLE_PYMOAB=ON \
+              -DENABLE_BLASLAPACK=OFF \
+              -DENABLE_FORTRAN=OFF \
+              -DCMAKE_INSTALL_PREFIX=$HOME/dagmc_bld/MOAB \
+              -DBUILD_SHARED_LIBS=ON \
+              -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib
```
Comment (pshriwise) [inline doc/install/dependencies.rst]: Small tweak here. MOAB doesn't specify a build type to CMake and can be _much_ slower without the optimization flags added with `-DCMAKE_BUILD_TYPE=Release`
```suggestion
    $ cmake ../moab -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
              -DCMAKE_BUILD_TYPE=Release \
              -DENABLE_PYMOAB=ON \
              -DENABLE_BLASLAPACK=OFF \
              -DENABLE_FORTRAN=OFF \
              -DCMAKE_INSTALL_PREFIX=$HOME/dagmc_bld/MOAB \
              -DBUILD_SHARED_LIBS=ON \
              -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib
```

File: doc/install/index.rst [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
 
-This document describes the DAGMC installation process. Before you begin, you
-should know that the following will be required:
+This document describes the DAGMC installation process. 
+
+If you plan to use OpenMC_, you can follow `their installation 
+instructions <OpenMC_install>`_ with
+DAGMC built-in.
+
+Otherwise.... Before you begin, you should know that the following will be
```
Comment (pshriwise) [inline doc/install/index.rst]:
```suggestion
Otherwise, before you begin, you should know that the following will be
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
Two small suggestions and I think that we're looking good here!

File: doc/install/dependencies.rst (hunk 4 above)
Comment (pshriwise) [reply doc/install/dependencies.rst]: Did these changes get dropped in the latest push by chance?

File: doc/index.rst [hunk 6]
```diff
@@ -50,13 +61,10 @@ materials and other geometry-related information.
 ..  _CNERG: https://cnerg.github.io
 ..  _MOAB: https://press3.mcs.anl.gov/sigma/moab-library
```
Comment (ahnaf-tahmid-chowdhury) [inline doc/index.rst]: I apologize for any inconvenience caused, but it seems that the previous link is no longer active. Please find the updated link provided below.

```suggestion
..  _MOAB: https://sigma.mcs.anl.gov/moab-library/
```

**Review by ahnaf-tahmid-chowdhury — CHANGES_REQUESTED** [review]

**Comment by gonuke** [comment]
This is failing windows tests @pshriwise - but it's probably not because of this PR... I've addressed your questions, so let me know if you think it's ready to merge.

**Comment by ahnaf-tahmid-chowdhury** [comment]
> This is failing windows tests @pshriwise - but it's probably not because of this PR... I've addressed your questions, so let me know if you think it's ready to merge.

Yes, I have also faced a similar problem in my PR #860. I am not sure, but I am assuming there is a bug in MOAB.

**Comment by makeclean** [comment]
Its a build failure due to Gtest bizzarely

```D:\a\DAGMC\DAGMC\src\gtest\gtest-1.8.0\src/gtest-internal-inl.h(641,31): error C2039: 'SetUpTestCaseFunc': is not a member of 'testing::Test' [D:\a\DAGMC\build\src\gtest\gtest.vcxproj]```

**Comment by ahnaf-tahmid-chowdhury** [comment]
> Its a build failure due to Gtest bizzarely
> 
> `D:\a\DAGMC\DAGMC\src\gtest\gtest-1.8.0\src/gtest-internal-inl.h(641,31): error C2039: 'SetUpTestCaseFunc': is not a member of 'testing::Test' [D:\a\DAGMC\build\src\gtest\gtest.vcxproj]`

Dear @makeclean, I have updated gtest to v1.13.0 and found the same error. Here is the [log](https://github.com/ahnaf-tahmid-chowdhury/DAGMC/actions/runs/5614293098/job/15212130671).

File: doc/install/index.rst [hunk 7]
```diff
@@ -1,17 +1,23 @@
...
+required:
 
 1.  A basic understanding of Unix/Linux shell commands and how to navigate
     the shell, for installing dependencies and DAGMC enabled codes
-2.  MCNP5_ or MCNP6_ source code, if you intend to install DAG-MCNP5 or
-    DAG-MCNP6
+2.  MCNP6_ source code, if you intend to install DAG-MCNP6 (or MCNP5_ source
+    code for DAG-MCNP5)
 3.  FLUKA_, if you intend to install FluDAG
 4.  Coreform Cubit_ for the creation of geometry
```
Comment (shimwell) [inline doc/install/index.rst:18]: Keen to see other tools mentioned for geometry creation but I might be a bit bias :-)

perhaps throw in a link to https://github.com/svalinn/DAGMC/discussions/812

**Comment by ahnaf-tahmid-chowdhury** [comment]
Dear @gonuke, I believe that some of the resolved changes might have been accidentally dropped in the latest force-push. Could you please check?

**Comment by gonuke** [comment]
> Dear @gonuke, I believe that some of the resolved changes might have been accidentally dropped in the latest force-push. Could you please check?

Oh no!  You may be right!  Let me see if I can recover them manually

File: doc/install/index.rst (hunk 7 above)
Comment (gonuke) [reply doc/install/index.rst:18]: I agree @shimwell , but think we need to develop a better strategy than just a link to #812 - feel free to PR some doc updates that address this.

Comment (shimwell) [reply doc/install/index.rst:18]: Thanks for being open to this, Once I've got that [cad-to-dagmc ](https://github.com/fusion-energy/cad_to_dagmc) passing a decent range of geometry comparisons over on the [benchmark zoo](https://github.com/fusion-energy/model_benchmark_zoo) then I shall be back with a PR to add a code to the list

**Comment by gonuke** [comment]
Converted this to draft until #880 is merged

**Comment by gonuke** [comment]
This has been rebased & tested (over-tested for now since it's only a doc change) and ready to merge once @pshriwise approves changes.

File: tmp
Comment (pshriwise) [inline tmp:1]: Should this be here?

**Review by pshriwise — CHANGES_REQUESTED** [review]
Thanks for diving into documentation updates, @gonuke!

What do you think about instructing users to rely on a package manager installation of HDF5 via `apt` or  `conda` rather than recommending building from source?

File: tmp
Comment (gonuke) [reply tmp:1]: Nope - definitely not!!!

File: doc/install/dependencies.rst [hunk 8]
```diff
@@ -79,45 +79,32 @@ Redhat linux users can do likewise with:
 MOAB installation
 ~~~~~~~~~~~~~~~~~
 
-As of DAGMC version 3.1, MOAB version 5.1.0 or higher is required. The following
+As of DAGMC version 3.2, a MOAB version between 5.1.0 and 5.3.0 is required. The following
```
Comment (pshriwise) [inline doc/install/dependencies.rst:82]: Why does this need to be between `5.10` and `5.3.0` again? I thought we could also support newer versions.

**Review by pshriwise — COMMENTED** [review]
One small quibble...

File: doc/install/dependencies.rst (hunk 8 above)
Comment (ahnaf-tahmid-chowdhury) [reply doc/install/dependencies.rst:82]: There are significant changes in v5.4.0 that are causing some [build fail](https://github.com/svalinn/DAGMC/pull/860#issuecomment-1541352060). We need to resolve this issue before proceeding to v5.4 MOAB.

Comment (pshriwise) [reply doc/install/dependencies.rst:82]: Ah, I see. Perhaps we could clarify that the restriction to 5.4.0 is only for Windows if that is the case?

Comment (gonuke) [reply doc/install/dependencies.rst:82]: Our CI is stuck on 5.3.0 until we can figure out all the issues.  It's a limitation for Ubuntu as well for 20.04 and older - seems to work on 22.04, though.

With all those caveats, it's seems best to stick with 5.3.0 for now

Comment (pshriwise) [reply doc/install/dependencies.rst:82]: Fair enough! Strange though... hopefully we can figure it out someday.

**Review by pshriwise — APPROVED** [review]

**Comment by pshriwise** [comment]
@ahnaf-tahmid-chowdhury I'm happy with this if you are. If you don't have any further comments, feel free to merge!

... if you have permissions, that is. Otherwise I can take care of it.

**Comment by ahnaf-tahmid-chowdhury** [comment]
I'm also satisfied with it. After my most recent active PR gets merged, I'm considering creating a new PR to address the latest version of MOAB. It appears that I inadvertently created a forked repo from another fork last time.

**Pull request merged**